File: src/02.string/01.base/03.14.ts
export function longestCommonPrefix(strList: string[]): string {
    let flag = false;
    let res = "";
    for (let i = 0; i < strList[0].length; i++) {
        for (let j = 1; j < strList.length; j++) {
            if (i >= strList[0].length || strList[j][i] !== strList[0][i]) {
                flag = true;
                break;
            }
        }
        if (flag) {
            break;
        }
        res += strList[0][i];
    }

    return res;
}

File: src/02.string/01.base/03.14.go
package base

func LongestCommonPrefix(strList []string) string {
	/*
		res := ""
		flag := false

		for i := 0; i < len(strList[0]); i++ {
			for j := 1; j < len(strList); j++ {
				if i >= len(strList[j]) || strList[j][i] != strList[0][i] {
					flag = true
					// strList[0] = strList[0][:i]
					break
				}
			}
			if flag {
				break
			}
			res = res + string(strList[0][i])
		}

		return res
	*/

	flag := false

	for i := 0; i < len(strList[0]); i++ {
		for j := 1; j < len(strList); j++ {
			if i >= len(strList[j]) || strList[j][i] != strList[0][i] {
				flag = true
				strList[0] = strList[0][:i]
				break
			}
		}
		if flag {
			break
		}
	}

	return strList[0]
}
